### ISSUE TITLE
Getting white skull after death when someone is getting dmg ticks you caused

### ISSUE THREAD

**Issue by gerotib** [opentibiabr/otservbr-global-archived#907]
Im not exactly sure how it does work on rl, but definitely you cant get skulled in pz in any way, so:
- player can get white skull after death (in temple or outside), if another player is still getting damage ticks from fields or utori caused by first player before he died. Additionally, without party protection on server, you can get white skulled when only party members were walking on your fields before death.

Video examples: (both with `ADVANCED_SECURE_MODE` and `PARTY_PROTECTION` off)
- [players not in party](https://gfycat.com/vastevergreenhummingbird) 
- [both players in party](https://gfycat.com/agitatedskinnycivet)

Steps to reproduce: (2nd vid)
1. Set `ADVANCED_SECURE_MODE` and `PARTY_PROTECTION` in global.lua to 0
2. Login two players
3. Create a party
4. Player 1 throws fire bomb (or any other field)
5. Player 2 steps on it, and kill Player 1 (Player 1 can also die to a monster)
6. Player 1 log in, and gets white skull the next time that Player 2 is getting ticked

Ingame example: (retro-pvp so party protection and advanced secure mode are off)
You hunt in party at monsters where you use energy (or any other) bombs, so monsters cant walk in. Your party members running on these fields, you accidentally die. You login in temple, run outside and get white skull out of nowhere. You cant get rid of this skull until your party member loses electrify.

Also with `ADVANCED_SECURE_MODE` and  `PARTY_PROTECTION` on this can happen, you just need to turn your secure mode off after you dies. (with secure mode on you wont get skull, because another player is getting "puffs" instead of damage)

I have no idea how to fix that, so cant really help :<

**Comment by Lycanzito** [comment]
this pull request, should fix it;

https://github.com/otland/forgottenserver/pull/2815

**Comment by gerotib** [comment]
@Lycanzito its related to another issue, but well also problem with fields and conditions. This pr you linked is to fix bug when you getting soulfired, and then you can walk on fire fields without taking damage, not to "clear" player that inflict someones burning when he dies.

**Comment by Lycanzito** [comment]
> @Lycanzito its related to another issue, but well also problem with fields and conditions. This pr you linked is to fix bug when you getting soulfired, and then you can walk on fire fields without taking damage, not to "clear" player that inflict someones burning when he dies.

fu#k, my bad, didnt pay much attention

**Comment by gerotib** [comment]
I found that its caused by #681: playerAutoID change that was responsible for saving client options correctly. (it was not an issue on vanilla tfs)

"Owner" of condition and thrown field is checked by `getCreatureByID` that is checking `playerAutoID` which is set on each character login. Before #681 `playerAutoID` was auto incremented (each login), so when player died and log again, he got another ID than before death. Now `playerAutoID` is "constant" (set by adding `guid` to default id value), so when player dies, and log again, doDamage function still gets him as the same "owner" of condition and thrown field. Thats why player will get skulled again.

Some links of what I am talking about:
- Code that gets "owner" (attacker) by playerAutoID: (same every player login after #681)
-- condition owner:
https://github.com/opentibiabr/otservbr-global/blob/e0045450ba35ae99ff5cf186ef6df6fe6ca9fe4d/src/condition.cpp#L1123
-- field owner:
https://github.com/opentibiabr/otservbr-global/blob/9e5e5321d2dc664f5ddbafb5d2a585c578c53185/src/combat.cpp#L1452

- `getCreatureByID` function itself:
https://github.com/opentibiabr/otservbr-global/blob/522507644143b654d27b74254e8478134f0f8410/src/game.cpp#L382

- function that set `playerAutoID` on each login:
https://github.com/opentibiabr/otservbr-global/blob/522507644143b654d27b74254e8478134f0f8410/src/player.h#L146

Not sure how it can be fixed, maybe something like separate "playerSessionID" that would be set like playerAutoID before (auto incremented, so unique each player login) and checked in this "get owner" functions. Or actually clearing "ownership" of these conditions and fields when player dies.

**Comment by heberholanda** [comment]
Fixed in "PK on PZ #946 "

**Comment by mrianura** [comment]
lol It's not fixed in "PK on PZ"
